lib/rubygems/uninstaller.rb

```diff
@@ -100,7 +100,11 @@ def uninstall
     end
 
     default_specs.each do |default_spec|
-      say "Gem #{default_spec.full_name} cannot be uninstalled because it is a default gem"
+      say(
+        "#{default_spec.full_name} is a default gem. If there is a user " \
+          "copy, it will be uninstalled, but the default copy cannot be " \
+          "uninstalled."
+      )
```

Comment:
Instead of adding conditionals to the message, since we can figure out whether there is a user copy or not, I would leave the message intact if there's no user copy, and don't print anything otherwise (because we'll give a message after we're doing with uninstall clarifying that case). So something like (completely untested):

```suggestion
       say "Gem #{default_spec.full_name} cannot be uninstalled because it is a default gem" unless list.include? {|s| s.full_name == default_spec.full_name }
```

Reply:
Good idea